engine/common/follower/compliance_core.go

```diff
@@ -58,15 +59,15 @@ var _ complianceCore = (*ComplianceCore)(nil)
 func NewComplianceCore(log zerolog.Logger,
 	mempoolMetrics module.MempoolMetrics,
 	heroCacheCollector module.HeroCacheMetrics,
-	finalizationConsumer hotstuff.FinalizationConsumer,
+	followerConsumer hotstuff.ConsensusFollowerConsumer,
 	state protocol.FollowerState,
 	follower module.HotStuffFollower,
 	validator hotstuff.Validator,
 	sync module.BlockRequester,
 	tracer module.Tracer,
 ) (*ComplianceCore, error) {
 	onEquivocation := func(block, otherBlock *flow.Block) {
-		finalizationConsumer.OnDoubleProposeDetected(model.BlockFromFlow(block.Header), model.BlockFromFlow(otherBlock.Header))
+		followerConsumer.OnDoubleProposeDetected(model.BlockFromFlow(block.Header), model.BlockFromFlow(otherBlock.Header))
```

Comment:
While reviewing the PR, I ran again in the issue that is was very hard to figure out where events are going and where the are triggered. In my experience, it becomes significantly easier, if we avoid anonymous functions for passing and consuming events. 

In this particular case here, what is the reason to not hand the `hotstuff.FollowerConsumer` directly to the cache? To me, this would this additional layer of obfuscation.

Reply:
The primary reason is for `Cache` to be more generic, but I can change it.

Reply:
the cache here is already very specific to the follower. We can always generalize it should we use the cache somewhere else ... honestly though, I think the chances of us using the cache elsewhere is reasonably low in my opinion. Therefore, I think airing on specificity to improve the code's clarity is in my opinion a good tradeoff.